src/state/observeSpeaker.test.ts

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2024 New Vector Ltd.
+
+SPDX-License-Identifier: AGPL-3.0-only
+Please see LICENSE in the repository root for full details.
+*/
+
+import { describe, test } from "vitest";
+import { distinctUntilChanged } from "rxjs";
+
+import { withTestScheduler } from "../utils/test";
+import { observeSpeaker } from "./observeSpeaker";
+
+const yesNo = {
+  y: true,
+  n: false,
+};
+
+describe("observeSpeaker", () => {
+  describe("does not activate", () => {
+    const expectedOutputMarbles = "n";
+    test("no speaking", () => {
+      const speakingInputMarbles = " n";
+      withTestScheduler(({ hot, expectObservable }) => {
+        expectObservable(
+          observeSpeaker(hot(speakingInputMarbles, yesNo)).pipe(
+            distinctUntilChanged(),
+          ),
+        ).toBe(expectedOutputMarbles, yesNo);
+      });
+    });
+
+    test("speaking for 1ms", () => {
+      const speakingInputMarbles = " y n";
+      withTestScheduler(({ hot, expectObservable }) => {
+        expectObservable(
+          observeSpeaker(hot(speakingInputMarbles, yesNo)).pipe(
+            distinctUntilChanged(),
+          ),
+        ).toBe(expectedOutputMarbles, yesNo);
+      });
+    });
+
+    test("speaking for 999ms", () => {
+      const speakingInputMarbles = " y 999ms n";
+      withTestScheduler(({ hot, expectObservable }) => {
+        expectObservable(
+          observeSpeaker(hot(speakingInputMarbles, yesNo)).pipe(
+            distinctUntilChanged(),
+          ),
+        ).toBe(expectedOutputMarbles, yesNo);
+      });
+    });
+
+    test("speaking intermittently", () => {
+      const speakingInputMarbles =
+        " y 199ms n 199ms y 199ms n 199ms y 199ms n 199ms y 199ms n 199ms y 199ms n 199ms y 199ms n 199ms y 199ms n 199ms y 199ms n";
```

Comment:
We should at least add one test that does multiple `y` in a row. I dont think this can happen but one never knows if there is some diamond'ing happening so we end up in a case where `speakingInputMarbles` emit true twice.

Reply:
Added in https://github.com/element-hq/element-call/commit/7f2a7b00abdeb999e05f6c4a6619dcca384e05a6